tools/ou3_p4_nextgen_gain_certificate.py

```
@@ -0,0 +1,44 @@
+ if any(v>Kg for v in K.values()): raise RuntimeError(f'{mode}: class-local K exceeds global')
+ h=base['directional_measurement_operator_norm_upper']; L={k:LEGACY.mul_up(K[k],float(h[k])) for k in K}; q=float(base['correction_quadratic_bound']['design_error_norm_radius']); Cva=float(base['vector_residual_quadratic_constant_acc_upper']); Cvm=float(base['vector_residual_quadratic_constant_mag_upper']); Cia=LEGACY.mul_up(K['accelerometer'],Cva); Cim=LEGACY.mul_up(K['magnetometer'],Cvm)
+ cs=LEGACY._composition_quadratic_constant(L['S_zero'],0.,q); ca=LEGACY._composition_quadratic_constant(L['accelerometer'],Cia,q); cm=LEGACY._composition_quadratic_constant(L['magnetometer'],Cim,q); C={'prediction':float(base['directional_operation_quadratic_defect_constants_upper']['prediction']),'S_zero_accepted':float(cs['full_state_quadratic_defect_constant_upper']),'accelerometer_accepted':float(ca['full_state_quadratic_defect_constant_upper']),'magnetometer_accepted':float(cm['full_state_quadratic_defect_constant_upper'])}; s=LEGACY.add_up(LEGACY.add_up(C['prediction'],C['S_zero_accepted']),LEGACY.add_up(C['accelerometer_accepted'],C['magnetometer_accepted'])); prev=float(base['directional_operation_defect_sum_per_sample_upper']);
+ if s>prev: raise RuntimeError(f'{mode}: defect sum regressed')
+ n=int(base['word_samples_upper']); mmin=float(base['metric_lambda_min_lower']); mmax=float(base['metric_lambda_max_upper']); B=LEGACY.div_up(LEGACY.mul_up(LEGACY.mul_up(LEGACY.mul_up(LEGACY.PREFIX_BOOTSTRAP_W_FACTOR,float(n)),LEGACY.sqrt_up(mmax)),s),mmin); Bprev=float(base['transported_word_defect_B_upper']);
+ if B>Bprev: raise RuntimeError(f'{mode}: B regressed')
+ gap=P4D._endpoint_sqrt_gap_lower(float(base['P3_word_endpoint_delta_lower'])); sw=min(LEGACY.div_down(gap,B),LEGACY.div_down(1.,B)); W=LEGACY.mul_down(sw,sw); Wprev=float(base['certified_level_W']); swprev=float(base['certified_level_sqrt_W']);
+ if W<Wprev: raise RuntimeError(f'{mode}: W regressed')
+ qp=LEGACY.mul_up(2.,LEGACY.sqrt_up(LEGACY.div_up(W,mmin)))
+ if not qp<q:
+  sw=swprev; W=Wprev; qp=float(base['prefix_canonical_error_norm_upper'])
+ corr={'S_zero':LEGACY.mul_up(L['S_zero'],qp),'accelerometer':LEGACY.add_up(LEGACY.mul_up(L['accelerometer'],qp),LEGACY.mul_up(Cia,qp*qp)),'magnetometer':LEGACY.add_up(LEGACY.mul_up(L['magnetometer'],qp),LEGACY.mul_up(Cim,qp*qp))}; cp=max(corr.values())
```

Ruff (0.16.2)

[error] Multiple statements on one line (semicolon)

(E702)